hikari/files.py

```diff
@@ -470,6 +480,40 @@ async def read(
         async with self.stream(executor=executor) as reader:
             return await reader.read()
 
+    async def save(
+        self,
+        path: typing.Optional[Pathish] = None,
+        *,
+        executor: typing.Optional[concurrent.futures.Executor] = None,
+        force: bool = False,
+    ) -> None:
+        """Save this resource to disk.
+
+        This writes the resource file in chunks, and so does not load
+        the entire resource into memory.
+
+        Parameters
+        ----------
+        path : typing.Optional[Pathish]
+            The path to save this resource to. If this is a string, the
+            path will be relative to the current working directory. If
+            `None`, the resource will be saved as `filename` in
+            the current working directory.
+        executor : typing.Optional[concurrent.futures.Executor]
+            The executor to run in for blocking operations.
+            If `None`, then the default executor is used for
+            the current event loop.
+        force : bool
+            Whether to overwrite an existing file. Defaults to `False`.
+        """
+        loop = asyncio.get_running_loop()
+        path = ensure_path(path or self.filename)
+
+        with await loop.run_in_executor(executor, _open_write_path, path, self.filename, force) as f:
+            async with self.stream(executor=executor) as reader:
+                async for chunk in reader:
+                    await loop.run_in_executor(executor, f.write, chunk)
```

Comment:
This could prob also be specialised for the Bytes impl (as an optimisation) to just save it all to the file at once instead of making multiple thread calls to chunk through it like the following
```py
class Bytes(...):
    async def _save_file(self, path: pathlib.Path, force: bool):
        _to_write_path(path, self.filename, force).write_bytes(self.data)

    async def save(self, path, executor, force):
        path = ensure_path(path or self.filename) 
        await loop.run_in_executor(executor, self._save_file, path, force)
```

Reply:
Again, thanks for the ideas!!